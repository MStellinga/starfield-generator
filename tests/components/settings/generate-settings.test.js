/* eslint-disable jest/valid-expect */
import { h } from 'preact';
import render from 'preact-render-to-string';
import GenerateSettings from '../../../src/components/settings/generate-settings';

const mockSettings =  {    
    nebulaBubbleBaseSize:30,
    nebulaBubbleMaxSize: 160,
    nebulaBubbleCenterBaseSize: 30,
    nebulaBubbleCenterMaxSize: 80,
    colorDistanceFalloff: 2.0,
    colorDampening: 4.0,
    fractalDivisionCount: 6,
    fractalMinSize: 5,
    fractalColorGain: 10,
    blueStars: false,
    blurNebula: false,
    blurFractal: false      
  }

describe('Test Generate settings', () => {
	test('Generate settings renders', () => {
        const tree = render(<GenerateSettings settings={{}} />);
        expect(tree).toMatchSnapshot();	
    });

    test('Generate settings renders with actual settings', () => {
        const tree = render(<GenerateSettings settings={mockSettings} />);
        expect(tree).toMatchSnapshot();	
    });   

    // Unfortunately, adding tests that call event handlers is hard to do with Enzyme and function-based Preact components (at this time, anyway)
});
