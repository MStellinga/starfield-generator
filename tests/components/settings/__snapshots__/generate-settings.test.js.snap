// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Generate settings Generate settings renders 1`] = `"<div class=\\"section\\"><table><tbody><tr><td>Bubble sizes</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" /></td><td>-</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" /></td><td colspan=\\"3\\">Fractal division count</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" /></td></tr><tr><td>Dark center size</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" /></td><td>-</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" /></td><td colspan=\\"3\\">Fractal min size</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" /></td></tr><tr><td colspan=\\"3\\">Bubble distance fall-off</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" /></td><td colspan=\\"3\\">Fractal color gain (%)</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" /></td></tr><tr><td colspan=\\"3\\">Bubble color desaturate</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" /></td><td colspan=\\"4\\"></td></tr><tr><td colspan=\\"4\\"><div class=\\"checkbox\\"><input type=\\"checkbox\\" /><span>Blur nebula</span></div></td><td colspan=\\"4\\"><div class=\\"checkbox\\"><input type=\\"checkbox\\" /><span>Blur fractal</span></div></td></tr></tbody></table></div>"`;

exports[`Test Generate settings Generate settings renders with actual settings 1`] = `"<div class=\\"section\\"><table><tbody><tr><td>Bubble sizes</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"30\\" /></td><td>-</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"160\\" /></td><td colspan=\\"3\\">Fractal division count</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"6\\" /></td></tr><tr><td>Dark center size</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"30\\" /></td><td>-</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"80\\" /></td><td colspan=\\"3\\">Fractal min size</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"5\\" /></td></tr><tr><td colspan=\\"3\\">Bubble distance fall-off</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"2\\" /></td><td colspan=\\"3\\">Fractal color gain (%)</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"10\\" /></td></tr><tr><td colspan=\\"3\\">Bubble color desaturate</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"4\\" /></td><td colspan=\\"4\\"></td></tr><tr><td colspan=\\"4\\"><div class=\\"checkbox\\"><input type=\\"checkbox\\" /><span>Blur nebula</span></div></td><td colspan=\\"4\\"><div class=\\"checkbox\\"><input type=\\"checkbox\\" /><span>Blur fractal</span></div></td></tr></tbody></table></div>"`;
